### DISCUSSION TITLE
Orgzly support of .org_archive files

### DISCUSSION THREAD

**Discussion by BrianA-MN** [orgzly-revived/orgzly-android-revived#255] [Q&A]
I have come to a point where I want to archive my old DONE items. I would like to keep both my orgzly repository and desktop repository in sync including the archive file.  But this currently can't be done in orgzly, because it apparently create and write to a .org file extension. I found this out by trial and error.  First create a .org file with org-mode and add some "TODO" and "DONE " tasks.  Then use the default archive function of org-mode to write to an archive file (not to move  to the end of existing file) with a file extension of .org_archive.  Now sysc the repository with orgzly and import the .org_archive.  What orgsly will do is create a new file in the repository location with the title of the org file you imported.  Move some tasks from the non-archive org file to the archive, using orgzly copy and passt. Those items are placed in the newly created file, but not in the .org_archive file of the orgzly repository.  Try to sync orgzly to the repository, then back to the desktop and find that the original .org file is modified, but not the .org_archive. 

 Org-mode archives to {filename}.org_archive (by default).  Unforunately orgzly doesn't recognize the file extension.  So the only way around this is to designate in the desktop org-mode an alternate "archive" file.  In org-mode this can be changed with the variable org-archive-location. Or by including the variable as #+ARCHIVE: file name:: in the file header.  I'm trying both methods to verify this works, but running into problems with the header variable format.

It would be nice if orgzly recognized the default .org_archive file. Any chance that can be added?  I'm not looking for the org-mode archive methodology, just the read/write ability to a file with a .org_archive extension.

Thank you.

**Comment by amberin** [comment]
Since Orgzly has no concept of "archiving", it makes sense to me that it is not aware of .org_archive files. What would it do with them? If we want Orgzly to support this concept, first we'd need to define the necessary user stories that the app should support, and how the UI should work. Should archive files appear as regular notebooks? Should they be synced by the app, but remain hidden and accessible through some special menu? I believe this type of questions neee to be addressed before dealing with the file types.

**Comment by BrianA-MN** [comment]
The issue of support for org-mode archiving actions had been discussed with the old orgzly application also.  I agree that the archive function would need to be addressed if orgzly-revived was trying to duplicate the org-mode archive function.  BUT that is not what is suggested here. The OP was asking if orgzly could recognize other file extensions and w/r to that imported notebook extension, in org mode format (which is simple structured text)? Or is there an impediment to reading and writing to non .org extension files?  Cheers.

**Reply by amberin** [reply]
I thought I replied to this at the time, but apparently I did not.

> is there an impediment to reading and writing to non .org extension files?

I can't say, because I don't understand the feature request. Do you want to be able to tell Orgzly to treat a specific file extension as if it were a .org file? That would not be very hard. But how useful is that? It would mean that all your org_archive files would appear as regular notebooks, and their names would be identical to their respective regular notebooks. This does not make sense to me, which is why I wonder if you are not in fact asking for a new type of notebook? And that is a much bigger change. So the user story needs clarification.
